File: VB/CustomStatusSample/AppointmentTemplates/VerticalAppointment.ascx.vb
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Drawing
Imports DevExpress.Web.ASPxScheduler.Internal
Imports DevExpress.Web.Internal
Imports DevExpress.XtraScheduler
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.HtmlControls
Imports System.Web.UI.WebControls

Namespace CustomStatusSample.AppointmentTemplates
    Partial Public Class VerticalAppointment
        Inherits System.Web.UI.UserControl

        Private ReadOnly Property Container() As VerticalAppointmentTemplateContainer
            Get
                Return CType(Parent, VerticalAppointmentTemplateContainer)
            End Get
        End Property
        Private ReadOnly Property Items() As VerticalAppointmentTemplateItems
            Get
                Return Container.Items
            End Get
        End Property

        Protected Overrides Sub OnLoad(ByVal e As EventArgs)
            MyBase.OnLoad(e)
            appointmentDiv.Style.Value = Items.AppointmentStyle.GetStyleAttributes(Page).Value
            horizontalSeparator.Style.Value = Items.HorizontalSeparator.Style.GetStyleAttributes(Page).Value

            lblStartTime.ControlStyle.MergeWith(Items.StartTimeText.Style)
            lblEndTime.ControlStyle.MergeWith(Items.EndTimeText.Style)
            lblTitle.ControlStyle.MergeWith(Items.Title.Style)
            lblDescription.ControlStyle.MergeWith(Items.Description.Style)

            PrepareImageContainer()
            Dim type As AppointmentStatusType = Container.AppointmentViewInfo.Status.Type
            If type = AppointmentStatusType.Tentative OrElse type = AppointmentStatusType.WorkingElsewhere Then
                SetStatusImage()
            Else
                statusDiv.Visible = False
                statusContainer.Controls.Add(Items.StatusControl)
            End If
            LayoutAppointmentImages()
        End Sub
        Private Sub SetStatusImage()
            If Container.AppointmentViewInfo.StatusDisplayType = AppointmentStatusDisplayType.Never Then
                statusDiv.Visible = False
            Else
                statusDiv.Visible = True
                tentativeStatusDiv.Style.Remove("background-image")
                tentativeStatusDiv.Style.Remove("height")
                tentativeStatusDiv.Style.Remove("top")
                If Container.AppointmentViewInfo.Status.Type = AppointmentStatusType.Tentative Then
                    tentativeStatusDiv.Style.Add("background-image", "url('Images/TentativeStatus.png')")
                End If
                If Container.AppointmentViewInfo.Status.Type = AppointmentStatusType.WorkingElsewhere Then
                    tentativeStatusDiv.Style.Add("background-image", "url('Images/WorkingElseWhereStatus.png')")
                End If
                tentativeStatusDiv.Style.Add("height", "98%")
                tentativeStatusDiv.Style.Add("top", "1px")
            End If
        End Sub
        Private Sub PrepareImageContainer()
            RenderUtils.SetTableSpacings(imageContainer, 1, 0)
        End Sub
        Private Sub PrepareImageCell(ByVal targetCell As HtmlTableCell)
            targetCell.Attributes.Add("class", "dxscCellWithPadding")
        End Sub
        Private Sub LayoutAppointmentImages()
            Dim count As Integer = Items.Images.Count
            For i As Integer = 0 To count - 1
                Dim row As New HtmlTableRow()
                Dim cell As New HtmlTableCell()
                PrepareImageCell(cell)
                AddImage(cell, Items.Images(i))
                row.Cells.Add(cell)
                imageContainer.Rows.Add(row)
            Next i
        End Sub
        Private Sub AddImage(ByVal targetCell As HtmlTableCell, ByVal imageItem As AppointmentImageItem)
            Dim image As New Image()
            imageItem.ImageProperties.AssignToControl(image, False)
            SchedulerWebEventHelper.AddOnDragStartEvent(image, ASPxSchedulerScripts.GetPreventOnDragStart())
            targetCell.Controls.Add(image)
        End Sub
    End Class
End Namespace

File: CS/CustomStatusSample/AppointmentTemplates/VerticalAppointment.ascx.cs
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Drawing;
using DevExpress.Web.ASPxScheduler.Internal;
using DevExpress.Web.Internal;
using DevExpress.XtraScheduler;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.HtmlControls;
using System.Web.UI.WebControls;

namespace CustomStatusSample.AppointmentTemplates
{
    public partial class VerticalAppointment : System.Web.UI.UserControl
    {
        VerticalAppointmentTemplateContainer Container { get { return (VerticalAppointmentTemplateContainer)Parent; } }
        VerticalAppointmentTemplateItems Items { get { return Container.Items; } }

        protected override void OnLoad(EventArgs e)
        {
            base.OnLoad(e);
            appointmentDiv.Style.Value = Items.AppointmentStyle.GetStyleAttributes(Page).Value;
            horizontalSeparator.Style.Value = Items.HorizontalSeparator.Style.GetStyleAttributes(Page).Value;

            lblStartTime.ControlStyle.MergeWith(Items.StartTimeText.Style);
            lblEndTime.ControlStyle.MergeWith(Items.EndTimeText.Style);
            lblTitle.ControlStyle.MergeWith(Items.Title.Style);
            lblDescription.ControlStyle.MergeWith(Items.Description.Style);

            PrepareImageContainer();
            AppointmentStatusType type = Container.AppointmentViewInfo.Status.Type;
            if (type == AppointmentStatusType.Tentative || type == AppointmentStatusType.WorkingElsewhere)
                SetStatusImage();
            else
            {
                statusDiv.Visible = false;
                statusContainer.Controls.Add(Items.StatusControl);
            }
            LayoutAppointmentImages();
        }
        void SetStatusImage()
        {
            if (Container.AppointmentViewInfo.StatusDisplayType == AppointmentStatusDisplayType.Never)
                statusDiv.Visible = false;
            else
            {
                statusDiv.Visible = true;
                tentativeStatusDiv.Style.Remove("background-image");
                tentativeStatusDiv.Style.Remove("height");
                tentativeStatusDiv.Style.Remove("top");
                if(Container.AppointmentViewInfo.Status.Type == AppointmentStatusType.Tentative)
                    tentativeStatusDiv.Style.Add("background-image", "url('Images/TentativeStatus.png')");
                if (Container.AppointmentViewInfo.Status.Type == AppointmentStatusType.WorkingElsewhere)
                    tentativeStatusDiv.Style.Add("background-image", "url('Images/WorkingElseWhereStatus.png')");
                tentativeStatusDiv.Style.Add("height", "98%");
                tentativeStatusDiv.Style.Add("top", "1px");
            }
        }
        void PrepareImageContainer()
        {
            RenderUtils.SetTableSpacings(imageContainer, 1, 0);
        }
        void PrepareImageCell(HtmlTableCell targetCell)
        {
            targetCell.Attributes.Add("class", "dxscCellWithPadding");
        }
        void LayoutAppointmentImages()
        {
            int count = Items.Images.Count;
            for (int i = 0; i < count; i++)
            {
                HtmlTableRow row = new HtmlTableRow();
                HtmlTableCell cell = new HtmlTableCell();
                PrepareImageCell(cell);
                AddImage(cell, Items.Images[i]);
                row.Cells.Add(cell);
                imageContainer.Rows.Add(row);
            }
        }
        void AddImage(HtmlTableCell targetCell, AppointmentImageItem imageItem)
        {
            Image image = new Image();
            imageItem.ImageProperties.AssignToControl(image, false);
            SchedulerWebEventHelper.AddOnDragStartEvent(image, ASPxSchedulerScripts.GetPreventOnDragStart());
            targetCell.Controls.Add(image);
        }	
    }
}